### DISCUSSION TITLE
How to use a pythonSoftIOC script with systemd?

### DISCUSSION THREAD

**Discussion by asarkar-parsys** [dls-controls/pythonSoftIOC#113] [Q&A] [answered]
Hi,

Has anyone tried launching pythonSoftIOC as a daemon with systemd?
My softioc script is based on the simple example you've provided, which defines the IOCs, the actions and then drops into interactive shell. This seems to work well when I launch the python script manually.
However, while launching the same via systemd as a daemon, systemd reports that the the service has been launched but it is in inactive (dead) state. On digging further, it looks like the script has reached the interactive shell but the process seems to exit, which is why systemd thinks that the service has completed. I have tried using TYPE for the service to be "oneshot" or "fork", however fork does not show up in the strace. If someone has successfully used it with systemd then it will be great if the .service file could be shared.

Is it possible to use Procserv while using pythonSoftIOC?

Thanks in advance.

Abhik

**Comment by Araneidae** [comment, marked as answer]
Yes, you should have no problem running your IOC under procServ, we do that all the time at DLS.  I imagine what is happening at the moment is that your IOC is being launched with stdin connected to `/dev/null`, is trying to read (because you've enabled the interactive shell), receiving EOF and terminating.  To be honest, this is as expected!

So either remove the interactive shell (if you're using cothread you can block with `cothread.WaitForQuit()` instead), or more usefully, run under procServ.

**Reply by Araneidae** [reply]
I'm going to assume this is answered!
